e2etest/zt_newe2e_oauth_device_test.go

```diff
@@ -0,0 +1,87 @@
+package e2etest
+
+import (
+	"flag"
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+	"time"
+)
+
+var runDeviceCodeTest = flag.Bool("device-code", false, "Whether or not to run device code tests. These must be run manually due to interactive nature.")
+
+func init() {
+	if runDeviceCodeTest != nil && *runDeviceCodeTest {
+		suiteManager.RegisterSuite(&DeviceLoginManualSuite{})
+	}
+}
+
+func Scenario_CopySync(svm *ScenarioVariationManager, env *AzCopyEnvironment) {
+	azCopyVerb := ResolveVariation(svm, []AzCopyVerb{AzCopyVerbCopy, AzCopyVerbSync}) // Calculate verb early to create the destination object early
+	// Scale up from service to object
+	dstObj := CreateResource[ContainerResourceManager](svm, GetRootResource(svm, ResolveVariation(svm, []common.Location{common.ELocation.Local(), common.ELocation.Blob(), common.ELocation.File()})), ResourceDefinitionContainer{}).GetObject(svm, "test", common.EEntityType.File())
+	// The object must exist already if we're syncing.
+	if azCopyVerb == AzCopyVerbSync {
+		dstObj.Create(svm, NewZeroObjectContentContainer(0), ObjectProperties{})
+
+		if !svm.Dryrun() {
+			// Make sure the LMT is in the past
+			time.Sleep(time.Second * 10)
+		}
+	}
+
+	body := NewRandomObjectContentContainer(svm, SizeFromString("10K"))
+	// Scale up from service to object
+	srcObj := CreateResource[ObjectResourceManager](svm, GetRootResource(svm, ResolveVariation(svm, []common.Location{common.ELocation.Local(), common.ELocation.Blob(), common.ELocation.File()})), ResourceDefinitionObject{
+		ObjectName: pointerTo("test"),
+		Body:       body,
+	})
+
+	// no local->local
+	if srcObj.Location().IsLocal() && dstObj.Location().IsLocal() {
+		svm.InvalidateScenario()
+		return
+	}
+
+	sasOpts := GenericAccountSignatureValues{}
+
+	stdout, _ := RunAzCopy(
```

Comment:
Can we validate that stdout is noting that we're connecting via AAD?

Reply:
I am not sure if we can get this information from stdout for validating if we are connected via AAD. Will check.

Reply:
We most definitely do log this information. Please check `credentialUtil.go`, it is logged by `logAuthType`. If it is not, and we are authorizing something, that is a bug.

Reply:
Done